src/main/java/pl/javastart/task/TournamentStats.java

```diff
@@ -1,12 +1,131 @@
 package pl.javastart.task;
 
-import java.util.Scanner;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
 
 public class TournamentStats {
 
+    private static final String STOP = "stop";
+    private static final String FIRSTNAME = "imię";
+    private static final String LASTNAME = "nazwisko";
+    private static final String SCORE = "wynik";
+    private static final int SORT_FIRSTNAME = 1;
+    private static final int SORT_LASTNAME = 2;
+    private static final int SORT_SCORE = 3;
+    private static final int ASCENDING = 1;
+    private static final int DESCENDING = 2;
+
     void run(Scanner scanner) {
+        ArrayList<Player> playersScore = getPlayersScore(scanner);
+        Comparator<Player> comparator = getPlayerComparator(scanner);
+        sortData(playersScore, comparator);
+        saveDataToFile(playersScore, "stats.csv");
+
         // tutaj dodaj swoje rozwiązanie
         // użyj przekazanego scannera do wczytywania wartości
 
     }
+
+    private Comparator<Player> getPlayerComparator(Scanner scanner) {
+        Comparator<Player> sortComparator = getSortComparator(scanner);
+        Comparator<Player> resultComparator = getPositiveOrNegativeComparator(scanner, sortComparator);
+        return resultComparator;
+    }
+
+    private Comparator<Player> getPositiveOrNegativeComparator(Scanner scanner, Comparator<Player> sortComparator) {
+        System.out.printf("Sortować rosnąco czy malejąco? (%s - rosnąco, %s - malejąco)\n", ASCENDING, DESCENDING);
+        Comparator<Player> sortComparator1 = sortComparator;
+        int option = inputInteger(scanner);
+        switch (option) {
+            case ASCENDING:
+                sortComparator1 = sortComparator;
+                break;
+            case DESCENDING:
+                sortComparator1 = sortComparator.reversed();
+                break;
+            default:
+                System.out.println("błędne wybór spsoobu sortowania");
+                System.exit(0);
```

Comment:
Stosowanie System.exit() jest najczęściej błędne, więcej info tutaj https://youtu.be/HsrEsJWwCSM